openedx/features/edly/tests/test_models.py

```diff
@@ -0,0 +1,113 @@
+"""
+Tests for Edly django models.
+"""
+import pytest
+from django.test import TestCase
+
+from openedx.features.edly.models import EdlyOrganization, EdlySubOrganization
+from openedx.features.edly.tests.factories import EdlyOrganizationFactory, EdlySubOrganizationFactory, SiteFactory
+
+
+class OrganizationTests(TestCase):
+    """
+    Tests for Organization Model.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(OrganizationTests, cls).setUpClass()
+
+    def test_organization_post_save_receiver(self):
+        """
+        Test entry is added to organization model and filter on it's value.
+        """
+        # add Organization entry to database
+        org_name = 'test org name'
+        org_slug = 'test org slug'
+        organization = EdlyOrganizationFactory(name=org_name, slug=org_slug)
+
+        # Verify an entry to Organization was added.
+        organizations = EdlyOrganization.objects.filter(name=org_name)
+
+        # Make sure an entry (and only one entry) is saved for Organization
+        assert len(organizations) == 1
+
+    def test_site_configuration_post_update_receiver(self):
+        """
+        Test that entry is updated in Organization model.
+        """
+        org_name = 'updated test org name'
+
+        # add SiteConfiguration to database
+        organization = EdlyOrganizationFactory()
+
+        organization.name = org_name
+        organization.save()
+
+        # Verify an entry to Organization was updated.
+        organizations = EdlyOrganization.objects.filter(name=org_name)
+
+        assert len(organizations) == 1
+        # Verify an entry to Organization was updated is filtered
+        assert organizations[0].name == org_name
+
+    def test_slug_alphanumerice_validation(self):
+        org_slug = 'test $%3org slug'
+        organization = EdlyOrganizationFactory(slug=org_slug)
+        expect_exception = 'Only alphanumeric characters are allowed.'
+
+        with pytest.raises(Exception) as excinfo:
+            organization.full_clean()
+
+        assert expect_exception in str(excinfo.value)
+
+
+class SubOrganizationTests(TestCase):
+    """
+    Tests for SubOrganization Model.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(SubOrganizationTests, cls).setUpClass()
+        cls.lms_site = SiteFactory()
+        cls.studio_site = SiteFactory()
+
+    def test_sub_organization_post_save_receiver(self):
+        """
+        Test that entry is added to SubOrganization model.
+        """
+        # add SubOrganization to database
```

Comment:
We need to be consistent with name `EdlyOrganization`, `EdlySubOrganization` and `Organization.

Reply:
Done